### ISSUE TITLE
[DNS Records]

### ISSUE THREAD

**Issue by robertwe** [Myra-Security-GmbH/terraform-provider-myrasec#14]
I have made a very simple test:
```
resource "myrasec_dns_record" "www" {
    domain_name = "robertwe.dev"
    name = "www.robertwe.dev"
    alternative_cname = "www-robertwe-dev.ax4z.com."
    record_type = "A"
    value = "192.168.0.1"
    ttl = 300
    active = true
    enabled = true
    upstream_options {
        backup = false
        down = false
        fail_timeout = 1
        max_fails = 100
        weight = 1
    }
}
```

Definition of the resource like below does not cause any problems and record is created properly but if I remove the optional property called `alternative_cname` Terraform is trying to create same record over and over. 

Here is the `terraform plan` : 
```
Terraform will perform the following actions:

  # myrasec_dns_record.www must be replaced
-/+ resource "myrasec_dns_record" "www" {
      - alternative_cname = "www-robertwe-dev.ax4z.com." -> null # forces replacement
      ~ created           = "2021-07-12T13:22:50+02:00" -> (known after apply)
      ~ id                = "190013" -> (known after apply)
      ~ modified          = "2021-07-12T13:22:50+02:00" -> (known after apply)
        name              = "www.robertwe.dev"
      - port              = 0 -> null
      - priority          = 0 -> null
      ~ record_id         = 190013 -> (known after apply)
        # (6 unchanged attributes hidden)

      ~ upstream_options {
          + created      = (known after apply)
          + modified     = (known after apply)
          ~ upstream_id  = 0 -> (known after apply)
            # (5 unchanged attributes hidden)
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

Side note: `name` property has to be a full canonical name otherwise we also got problem with the record recreation.

**Comment by robertwe** [comment]
fyi: Im currently working on a fix for that.

**Comment by robertwe** [comment]
quick update - first version of the fix is ready to go. Im going to make some additional tests just to be sure.

**Comment by SchlegelS0208** [comment]
Is this still valid or was this issue fixed by myrasec-provider version 1.18.0 ?

**Comment by xellio** [comment]
This issue should be fixed using the new released **v1.19.0** of the terraform provider.
